### PR TITLE
@alloy => Use root url if we cannot generate a public artsy url

### PR DESCRIPTION
**Pull request by orta**
When I looked through our crashes 
https://rink.hockeyapp.net/manage/apps/37029/app_versions/44/crash_reasons/54275149?type=crashes

I kept seeing the same thing, this always happens after onboarding, not sure the exact cause of it, but this should deal with the key issue of "it crashes."

### CODE REVIEW

File: Artsy/App/ARSpotlight.m [hunk 1]
```diff
@@ -72,7 +72,8 @@ + (BOOL)isSpotlightAvailable;
 
 + (NSURL *)webpageURLForEntity:(id<ARSpotlightMetadataProvider>)entity;
 {
-    return [[ARRouter baseDesktopWebURL] URLByAppendingPathComponent:entity.publicArtsyPath];
+    NSString *path = entity.publicArtsyPath ?: @"";
+    return [[ARRouter baseDesktopWebURL] URLByAppendingPathComponent:path];
```
Comment (alloy) [inline Artsy/App/ARSpotlight.m]: This means that various artworks could all get associated with the same base URL, which isn’t right, the URL should be the work’s unique identifier.

So the real reason needs to be found out and if there’s no possible fix for getting the right data at this time, then in those cases the work shouldn’t be indexed at all imo.

**Comment by orta** [comment]
Alright, so digging into this further, there is no way that the result of a `publicArtsyPath` could be nil

**Comment by orta** [comment]
All of the models that conform do something like:

```
- (NSString *)publicArtsyPath
{
    return [NSString stringWithFormat:@"/show/%@", self.showID];
}
```

Must be a nil getting in to the process somewhere.

**Comment by orta** [comment]
Looks like it's something related to genes, it's also very specifically on iPhone-y devices. Could be something to do with the genes we present in the list.

**Comment by orta** [comment]
Alright, so, they all exhibit an issue with networking at the start of their session. This means that [this call](https://github.com/artsy/eigen/blob/master/Artsy/View_Controllers/Login_and_Onboarding/AROnboardingViewController.m#L88) could fail. So, when it does fail, we fall back to [these gene](https://github.com/artsy/eigen/blob/master/Artsy/View_Controllers/Login_and_Onboarding/Onboarding_stages/5-_Onboarding_Personalization/ARPersonalizeViewController.m#L44):
```
            NSArray *fallbackGenes = @[ @"Photography", @"Bauhaus", @"Dada", @"Glitch Aesthetic", @"Computer Art", @"Op Art", @"Minimalism" ];
```

Problem is:

screen shot 2015-11-18 at 12 39 15 pm

So the fix is re-naming the gene.

File: Artsy/App/ARSpotlight.m (hunk 1 above)
Comment (orta) [inline Artsy/App/ARSpotlight.m]: I think this is fine, it takes something to be totally broken and not accessible on the API for it to break

**Comment by alloy** [comment]
Doh.

That all sounds very logical indeed. Knowing this, have you been able to reproduce the exact crash we were receiving?

**Comment by orta** [comment]
Yep

**Comment by alloy** [comment]
Ace

File: Artsy/App/ARSpotlight.m (hunk 1 above)
Comment (alloy) [inline Artsy/App/ARSpotlight.m]: Do you really want to still include this? As you showed before, the `publicArtsyPath` can never be `nil`. So this really seems like some noise that future readers could get confused by.

Comment (orta) [inline Artsy/App/ARSpotlight.m]: Nah, I can remove it.

Comment (alloy) [inline Artsy/App/ARSpotlight.m]: Jippie!

Comment (alloy) [inline Artsy/App/ARSpotlight.m]: I’ll do that after merging.

**Pull request merged**